crm_salesperson_planner/models/crm_salesperson_planner_visit_template.py

```diff
@@ -0,0 +1,406 @@
+        comodel_name="res.users",
+        string="Salesperson",
+        tracking=True,
+        default=lambda self: self.env.user,
+        domain=lambda self: [
+            ("groups_id", "in", self.env.ref("sales_team.group_sale_salesman").id)
+        ],
+    )
+    description = fields.Html(string="Description")
+    state = fields.Selection(
+        string="Status",
+        required=True,
+        copy=False,
+        tracking=True,
+        selection=[
+            ("draft", "Draft"),
+            ("in-progress", "In Progress"),
+            ("done", "Done"),
+            ("cancel", "Cancelled"),
+        ],
+        default="draft",
+    )
+    visit_ids = fields.One2many(
+        comodel_name="crm.salesperson.planner.visit",
+        inverse_name="visit_template_id",
+        string="Visit Template",
+    )
+    visit_ids_count = fields.Integer(
+        string="Number of Sales Person Visits", compute="_compute_visit_ids_count"
+    )
+    rrule = fields.Char(
+        string="Recurrent Rule",
+        compute="_compute_rrule",
+        inverse="_inverse_rrule",
+        store=True,
+    )
+    rrule_type = fields.Selection(
+        [
+            ("daily", "Days"),
+            ("weekly", "Weeks"),
+            ("monthly", "Months"),
+            ("yearly", "Years"),
+        ],
+        string="Recurrence",
+        default="daily",
+        states={"done": [("readonly", True)]},
+        required=True,
+        help="Let the visit automatically repeat at that interval",
+    )
+    end_type = fields.Selection(
+        [("count", "Number of repetitions"), ("end_date", "End date")],
+        string="Recurrence Termination",
+        default="count",
+    )
+    interval = fields.Integer(
+        string="Repeat Every", default=1, help="Repeat every (Days/Week/Month/Year)"
+    )
+    count = fields.Integer(string="Repeat", help="Repeat x times", default=1)
+    mo = fields.Boolean(string="Mon")
+    tu = fields.Boolean(string="Tue")
+    we = fields.Boolean(string="Wed")
+    th = fields.Boolean(string="Thu")
+    fr = fields.Boolean(string="Fri")
+    sa = fields.Boolean(string="Sat")
+    su = fields.Boolean(string="Sun")
+    month_by = fields.Selection(
+        [("date", "Date of month"), ("day", "Day of month")],
+        string="Option",
+        default="date",
+    )
+    day = fields.Integer(string="Date of month", default=1)
+    week_list = fields.Selection(
+        [
+            ("MO", "Monday"),
+            ("TU", "Tuesday"),
+            ("WE", "Wednesday"),
+            ("TH", "Thursday"),
+            ("FR", "Friday"),
+            ("SA", "Saturday"),
+            ("SU", "Sunday"),
+        ],
+        string="Weekday",
+    )
+    byday = fields.Selection(
+        [
+            ("1", "First"),
+            ("2", "Second"),
+            ("3", "Third"),
+            ("4", "Fourth"),
+            ("5", "Fifth"),
+            ("-1", "Last"),
+        ],
+        string="By day",
+    )
```

Comment:
We are working on a similar module, but it is based on calendar.event instead of creating a new data model from scratch, to take advantage of all the "magic" that Odoo has implemented for recurrence.
Why are you copying the methods used by calendar.event instead of inheriting it?

Reply:
Another advantage of using calendar.event directly is that all this information is directly displayed in each salesman's calendar and it can even be automatically synchronized with google calendar

Reply:
I continue this conversation in the other one already created.